From 7d85f4e56662d00a63190cd687280466452bfa84 Mon Sep 17 00:00:00 2001
Message-ID: <7d85f4e56662d00a63190cd687280466452bfa84.1707763691.git.sharlatanus@gmail.com>
In-Reply-To: <e6295f25eabb9fcc8765561825ab7c9b124c1568.1707763691.git.sharlatanus@gmail.com>
References: <e6295f25eabb9fcc8765561825ab7c9b124c1568.1707763691.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sat, 10 Feb 2024 21:32:16 +0000
Subject: [PATCH v2 26/30] gnu: go-golang-org-x-image: Move to golang-build.

* gnu/packages/golang.scm (go-golang-org-x-image): Move from here ...
* gnu/packages/golang-build.scm: ... to here.

* gnu/packages/game.scm: Add (gnu packages golang-build) module.

Change-Id: Icd0b303d8f7544f6eae10ac30d7bd01779abaede
---
 gnu/packages/games.scm        |  1 +
 gnu/packages/golang-build.scm | 31 +++++++++++++++++++++++++++++++
 gnu/packages/golang.scm       | 30 ------------------------------
 3 files changed, 32 insertions(+), 30 deletions(-)

diff --git a/gnu/packages/games.scm b/gnu/packages/games.scm
index f95681fbb7..c257a2cfe8 100644
--- a/gnu/packages/games.scm
+++ b/gnu/packages/games.scm
@@ -150,6 +150,7 @@ (define-module (gnu packages games)
   #:use-module (gnu packages gnupg)
   #:use-module (gnu packages gnuzilla)
   #:use-module (gnu packages golang)
+  #:use-module (gnu packages golang-build)
   #:use-module (gnu packages gperf)
   #:use-module (gnu packages graphics)
   #:use-module (gnu packages graphviz)
diff --git a/gnu/packages/golang-build.scm b/gnu/packages/golang-build.scm
index 87f184fb14..b717c3ad27 100644
--- a/gnu/packages/golang-build.scm
+++ b/gnu/packages/golang-build.scm
@@ -1,6 +1,7 @@
 ;;; GNU Guix --- Functional package management for GNU
 ;;; Copyright © 2019 Brian Leung <bkleung89@gmail.com>
 ;;; Copyright © 2020 Danny Milosavljevic <dannym@scratchpost.org>
+;;; Copyright © 2020 Efraim Flashner <efraim@flashner.co.il>
 ;;; Copyright © 2020 HiPhish <hiphish@posteo.de>
 ;;; Copyright © 2020 Leo Famulari <leo@famulari.name>
 ;;; Copyright © 2020 Oleg Pykhalov <go.wigust@gmail.com>
@@ -99,6 +100,36 @@ (define-public go-golang-org-x-exp
 @code{old} directory) packages.")
     (license license:bsd-3)))
 
+(define-public go-golang-org-x-image
+  (let ((commit "58c23975cae11f062d4b3b0c143fe248faac195d")
+        (revision "1"))
+    (package
+      (name "go-golang-org-x-image")
+      (version (git-version "0.0.0" revision commit))
+      (source
+       (origin
+         (method git-fetch)
+         (uri (git-reference
+               (url "https://go.googlesource.com/image")
+               (commit commit)))
+         (file-name (string-append "go.googlesource.com-image-"
+                                   version "-checkout"))
+         (sha256
+          (base32 "0i2p2girc1sfcic6xs6vrq0fp3szfx057xppksb67kliywjjrm5x"))))
+      (build-system go-build-system)
+      (arguments
+       `(#:import-path "golang.org/x/image"
+         ;; Source-only package
+         #:tests? #f
+         #:phases
+         (modify-phases %standard-phases
+           (delete 'build))))
+      (home-page "https://go.googlesource.com/image")
+      (synopsis "Supplemental Go image libraries")
+      (description "This package provides supplemental Go libraries for image
+processing.")
+      (license license:bsd-3))))
+
 (define-public go-golang-org-x-mod
   (let ((commit "7c05a442b7c1d1a107879b4a090bb5a38d3774a1")
         (revision "0"))
diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index 4c2781bf45..0437b4e83d 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -3488,36 +3488,6 @@ (define-public gopls
 editor.")
     (license license:bsd-3)))
 
-(define-public go-golang-org-x-image
-  (let ((commit "58c23975cae11f062d4b3b0c143fe248faac195d")
-        (revision "1"))
-    (package
-      (name "go-golang-org-x-image")
-      (version (git-version "0.0.0" revision commit))
-      (source (origin
-                (method git-fetch)
-                (uri (git-reference
-                       (url "https://go.googlesource.com/image")
-                       (commit commit)))
-                (file-name (string-append "go.googlesource.com-image-"
-                                          version "-checkout"))
-                (sha256
-                 (base32
-                  "0i2p2girc1sfcic6xs6vrq0fp3szfx057xppksb67kliywjjrm5x"))))
-      (build-system go-build-system)
-      (arguments
-       `(#:import-path "golang.org/x/image"
-         ; Source-only package
-         #:tests? #f
-         #:phases
-         (modify-phases %standard-phases
-           (delete 'build))))
-      (home-page "https://go.googlesource.com/image")
-      (synopsis "Supplemental Go image libraries")
-      (description "This package provides supplemental Go libraries for image
-processing.")
-      (license license:bsd-3))))
-
 (define-public go-golang-org-x-oauth2
   (let ((commit "0f29369cfe4552d0e4bcddc57cc75f4d7e672a33")
         (revision "1"))
-- 
2.41.0

